### PR TITLE
Cycle Sort in Python

### LINKED ISSUE #979: Cycle Sort in Python

### PR DESCRIPTION
**Pull request by Bhairavi-shah**
Fixes #979 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Sample Input / Output.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Changes proposed in this pull request:
- Cycle Sort in Python

#### Languages Used:
- Python

#### Files Added:
- CycleSort/Cycle_Sort.py

### CODE REVIEW

**Comment by Bhairavi-shah** [comment]
Hey,

I actually learnt it from there and wrote it. Hence the similarity. I will
work on it again.

**Comment by Bhairavi-shah** [comment]
Have Scraped out the previous code and re-written on my own fully. Please review my code.

File: CycleSort/Cycle_Sort.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
+def cycleSort(array, size):
```
Comment (marwahmanbir) [inline CycleSort/Cycle_Sort.py]: Good work! Just some minor changes and we're good to go.

File: CycleSort/Cycle_Sort.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    for start in range(0,n-1): #Put array element in the correct place
+        key = array[start]
+        location = start
+        for i in range(start+1, n): #Count smaller element in the right side of key
+            if(array[i] < key):
```
Comment (marwahmanbir) [inline CycleSort/Cycle_Sort.py]: Space after ``if``.

File: CycleSort/Cycle_Sort.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+            while key == array[location]: #If same data is found, increase location
+                location += 1
+            
+            if key != array[location]:
+                array[location], key = key, array[location]
+
+#Main program
+a = [23, 63, 98, 74, 20, 14, 36, 45, 99, 78, 98]
+n = len(a)
+cycleSort(a,n)
```
Comment (marwahmanbir) [inline CycleSort/Cycle_Sort.py]: Space before **n**.

File: CycleSort/Cycle_Sort.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+print("After sort : ")
+for i in range(0,n):
```
Comment (marwahmanbir) [inline CycleSort/Cycle_Sort.py]: Space before **n**.

File: CycleSort/Cycle_Sort.py [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    print(a[i], end=" ")
```
Comment (marwahmanbir) [inline CycleSort/Cycle_Sort.py]: Space around **=**.

**Pull request merged**